### PR TITLE
[google_maps] Endorses package:google_maps_flutter_web.

### PR DESCRIPTION
**Pull request by ditman**
This PR endorses `package:google_maps_flutter_web` as the implementation of `google_maps_flutter` for the web platform, and updates the relevant documentation.

### Issues

* Fixes https://github.com/flutter/flutter/issues/80688

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by ditman** [comment]
This PR should cause that the integration tests of `google_maps` run on `platform:web`. AFAIK we don't have specific tests for package endorsement.

**Comment by ditman** [comment]
```
�[31mThe license block for these files is missing or incorrect:�[0m
�[31m  /tmp/cirrus-ci-build/packages/google_maps_flutter/google_maps_flutter/example/web/index.html�[0m
�[31mIf this third-party code, move it to a "third_party/" directory, otherwise ensure that you are using the exact copyright and license text used by all first-party files in this repository.
```

Whoooops!

**Comment by stuartmorgan** [comment]
> This PR should cause that the integration tests of `google_maps` run on `platform:web`.

And it does!

Huh, do web integration tests run by `flutter drive` just not output anything about the tests being run? I'd never looked closely at the web output before, but I see that everything is that way.

Is there a way to get `flutter drive` to list the tests that are run?

> AFAIK we don't have specific tests for package endorsement.

Yes, adding web support to the example is the test.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.3.0
+
+* Endorses `package:google_maps_flutter_web` as the web implementation of this plugin.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: Nit: we usually just use the package name, not `package:`, when referring to a package.

File: packages/google_maps_flutter/google_maps_flutter/README.md [hunk 2]
```diff
@@ -98,7 +101,16 @@ import GoogleMaps
 }
 ```
 
-### Both
+### Web
+
+You'll need to modify the `web/index.html` file of your Flutter Web application
+to include the Google Maps JS SDK.
+
+Check [package:google_maps_flutter_web README](https://pub.dev/packages/google_maps_flutter_web)
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/README.md:109]: How about "The `google_maps_flutter_web` README" as the link text.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits!

**Comment by ditman** [comment]
I'll address the nits before landing this, thanks for the review!

> Huh, do web integration tests run by flutter drive just not output anything about the tests being run? I'd never looked closely at the web output before, but I see that everything is that way.

@stuartmorgan They normally only output some information about the compilation, and the "All Tests Passed" line. If there's an error, they'll spit the failing test and the stack trace.

I've only seen the normal "progress" that other tests have when running the tests manually in the browser (by `flutter run`ning the test file), but I suspect `flutter drive` is eating the browser console output (maybe?).

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md (hunk 1 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:3]: Removed "package:", and added a link to the package in pub, to maximize the chances that people find its readme :)

File: packages/google_maps_flutter/google_maps_flutter/README.md (hunk 2 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter/README.md]: Yes, I like it better, also added the backticks around the package name to make it clearer.

**Pull request merged**